src/catalog/src/remote/manager.rs

```diff
@@ -532,12 +532,51 @@ impl CatalogList for RemoteCatalogManager {
 
     /// List all catalogs from metasrv
     fn catalog_names(&self) -> Result<Vec<String>> {
-        Ok(self.catalogs.load().keys().cloned().collect::<Vec<_>>())
+        Ok(self.catalogs.iter().map(|k| k.key().to_string()).collect())
     }
 
     /// Read catalog info of given name from metasrv.
     fn catalog(&self, name: &str) -> Result<Option<CatalogProviderRef>> {
-        Ok(self.catalogs.load().get(name).cloned())
+        let maybe_catalog = self.catalogs.get(name);
+
+        if let Some(catalog) = maybe_catalog {
+            return Ok(Some(catalog.clone()));
+        }
+
```

Comment:
we can add some comments to explain this behaviour. It's for lack of incremental catalog syncing between datanode and meta. Here we fetch catalog from meta on demand. This can be removed when incremental catalog syncing is done in datanode.